### PR TITLE
Add event names for logs (#6818)

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/Mvc/issues/6818

### CODE REVIEW

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/MvcCoreLoggerExtensions.cs [hunk 1]
```diff
@@ -156,498 +156,499 @@ static MvcCoreLoggerExtensions()
...
+                new EventId(27, "DoneAttemptingToValidateParameter"),
                 "Done attempting to validate the bound parameter '{ParameterName}' of type '{ModelType}'.");
 
             _noNonIndexBasedFormatFoundForCollection = LoggerMessage.Define<string, string>(
                 LogLevel.Debug,
-                28,
+                new EventId(28, "NoNonIndexBasedFormatFoundForCollection"),
                 "Could not bind to collection using a format like {ModelName}=value1&{ModelName}=value2");
 
+
```
Comment (dougbu) [inline src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/MvcCoreLoggerExtensions.cs]: Don't need this extra blank line

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/MvcCoreLoggerExtensions.cs [hunk 2]
```diff
@@ -156,498 +156,499 @@ static MvcCoreLoggerExtensions()
...
             _actionMethodExecuted = LoggerMessage.Define<string, string, double>(
                 LogLevel.Information,
-                2,
+                new EventId(2, "ActionMethodExecuted"),
                 "Executed action method {ActionName}, returned result {ActionResult} in {ElapsedMilliseconds}ms.");
 
             _logFilterExecutionPlan = LoggerMessage.Define<string, string[]>(
                 LogLevel.Debug,
-                1,
+                new EventId(1, "LogFilterExecutionPlan"),
```
Comment (dougbu) [inline src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/MvcCoreLoggerExtensions.cs]: This name doesn't look like any of the others. Suggest `FilterExecutionPlan` or `OrderedFilterExecutionPlan`.

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Razor/MvcRazorLoggerExtensions.cs [hunk 3]
```diff
@@ -36,87 +36,82 @@ static MvcRazorLoggerExtensions()
...
-                4,
+                new EventId(4, nameof(ViewCompilerNoCompiledViewsFound)),
                 "Initializing Razor view compiler with no compiled views.");
 
             _viewCompilerLocatedCompiledViewForPath = LoggerMessage.Define<string>(
                 LogLevel.Trace,
-                5,
-                "Located compiled view for view at path '{Path}'.");
-
-            _viewCompilerLocatedCompiledViewForPath = LoggerMessage.Define<string>(
```
Comment (dougbu) [inline src/Mvc/src/Microsoft.AspNetCore.Mvc.Razor/MvcRazorLoggerExtensions.cs]: 👍

**Review by dougbu — APPROVED** [review]
How visible will these new names be? If they'll appear in the output, we might want to be more consistent about `Starting`&hellip; -> `Done`&hellip; versus `Before`&hellip; -> `After`&hellip; or other conventions we've used. But, I don't see a need to block this PR for decisions on the _right_ convention.

**Comment by pranavkm** [comment]
> How visible will these new names be?

From https://github.com/aspnet/Mvc/issues/6818#issuecomment-410024636, the names appear in the log output and used for filtering. I guess it's important that it's unique, but the issue doesn't say much about the nomenclature.

**Review by dougbu — APPROVED** [review]
Looking even better 😸

**Comment by pranavkm** [comment]
@Kahbazi thanks for the PR

**Comment by BrennanConroy** [comment]
FYI, we try to avoid using `nameof` for the event names as changing them is "technically" a breaking change and it's easier to avoid accidentally changing the name by using strings instead of function names.
Couple examples: https://github.com/aspnet/AspNetCore/blob/321327f84b2b22dcff2e9beb06a9a64236c5cced/src/SignalR/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.Log.cs
and
https://github.com/aspnet/Extensions/blob/1878d16ecd5165849cd6a78ce708e499b71477ca/src/HttpClientFactory/Http/src/DefaultHttpClientFactory.cs#L337

**Comment by Kahbazi** [comment]
@pranavkm you're welcome. Should I remove the `nameof` as @BrennanConroy said?

**Comment by pranavkm** [comment]
> Should I remove the nameof as @BrennanConroy said?

Go for it!

**Pull request merged**